lib/matplotlib/cbook/__init__.py

```diff
@@ -1417,7 +1417,19 @@ def _reshape_2D(X, name):
     *name* is used to generate the error message for invalid inputs.
     """
     # Iterate over columns for ndarrays, over rows otherwise.
-    X = np.atleast_1d(X.T if isinstance(X, np.ndarray) else np.asarray(X))
+    # Delete masked points for MaskedArray
+    if isinstance(X, np.ma.MaskedArray):
+        if X.ndim == 2:
+            X = [(x[~x.mask] if np.ma.getmask(x).any() else x) for x in X]
```

Comment:
These expressions can be greatly simplified by using the `compressed()` method of masked arrays.  But see my general comment.